src/main/java/io/opentracing/contrib/global/GlobalSpanManager.java

```diff
@@ -0,0 +1,87 @@
+package io.opentracing.contrib.global;
+
+import io.opentracing.Span;
+
+import java.io.Closeable;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class manages the globally 'active' {@link Span} in the current Thread.
+ *
+ * @author Sjoerd Talsma
+ */
+public final class GlobalSpanManager {
+
+    private static final Logger LOGGER = Logger.getLogger(GlobalSpanManager.class.getName());
+
+    private static final ThreadLocal<ManagedSpan> ACTIVE_SPAN = new InheritableThreadLocal<ManagedSpan>();
```

Comment:
Note that this can be a somewhat polarizing decision. Eventhough thread locals are popular, they are easy to get lost. Plus there are folks on both sides of the "should it be inheritable or not" debate. You'll notice log4j2 for example has a flag as to whether it is inherited. The discussion is here https://issues.apache.org/jira/browse/LOG4J2-479

Moreover, there are folks who do not want to thread locals or want to bring their own. For example, google recently changed their Context object so that internal systems can interop with a different storage.
https://github.com/grpc/grpc-java/blob/master/context/src/main/java/io/grpc/Context.java#L125

Main thing is to be open about the tradeoffs here, especially if the goal is wide adoption.